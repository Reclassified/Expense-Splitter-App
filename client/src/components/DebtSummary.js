import React from 'react';

const DebtSummary = ({ creditors, debtors }) => (
  <>
    {creditors.length > 0 && (
      <div className="creditors-section">
        <h3>Who's Owed Money</h3>
        <div className="creditors-list">
          {creditors.map((creditor) => (
            <div key={creditor.userId} className="creditor-item">
              <span className="creditor-name">{creditor.username}</span>
              <span className="creditor-amount positive">
                +${creditor.netBalance.toFixed(2)}
              </span>
            </div>
          ))}
        </div>
      </div>
    )}
    {debtors.length > 0 && (
      <div className="debtors-section">
        <h3>Who Owes Money</h3>
        <div className="debtors-list">
          {debtors.map((debtor) => (
            <div key={debtor.userId} className="debtor-item">
              <span className="debtor-name">{debtor.username}</span>
              <span className="debtor-amount negative">
                -${Math.abs(debtor.netBalance).toFixed(2)}
              </span>
            </div>
          ))}
        </div>
      </div>
    )}
  </>
);

export default DebtSummary;
