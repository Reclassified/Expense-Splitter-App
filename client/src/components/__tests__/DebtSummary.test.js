import React from 'react';
import { render } from '@testing-library/react';
import DebtSummary from '../DebtSummary';

describe('DebtSummary', () => {
  const creditors = [
    { userId: 1, username: 'Alice', netBalance: 50.25 },
  ];
  const debtors = [
    { userId: 2, username: 'Bob', netBalance: -50.25 },
  ];

  it('renders correctly and matches snapshot', () => {
    const { container } = render(
      <DebtSummary creditors={creditors} debtors={debtors} />
    );
    expect(container).toMatchSnapshot();
  });

  it('shows correct creditor and debtor info', () => {
    const { getByText } = render(
      <DebtSummary creditors={creditors} debtors={debtors} />
    );
    expect(getByText("Who's Owed Money")).toBeInTheDocument();
    expect(getByText('Alice')).toBeInTheDocument();
    expect(getByText('+$50.25')).toBeInTheDocument();
    expect(getByText('Who Owes Money')).toBeInTheDocument();
    expect(getByText('Bob')).toBeInTheDocument();
    expect(getByText('-$50.25')).toBeInTheDocument();
  });
}); 