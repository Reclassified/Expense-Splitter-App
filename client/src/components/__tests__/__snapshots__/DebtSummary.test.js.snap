// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DebtSummary renders correctly and matches snapshot 1`] = `
<div>
  <div
    class="creditors-section"
  >
    <h3>
      Who's Owed Money
    </h3>
    <div
      class="creditors-list"
    >
      <div
        class="creditor-item"
      >
        <span
          class="creditor-name"
        >
          Alice
        </span>
        <span
          class="creditor-amount positive"
        >
          +$
          50.25
        </span>
      </div>
    </div>
  </div>
  <div
    class="debtors-section"
  >
    <h3>
      Who Owes Money
    </h3>
    <div
      class="debtors-list"
    >
      <div
        class="debtor-item"
      >
        <span
          class="debtor-name"
        >
          Bob
        </span>
        <span
          class="debtor-amount negative"
        >
          -$
          50.25
        </span>
      </div>
    </div>
  </div>
</div>
`;
